src/main/java/org/jenkinsci/plugins/workflow/cps/CpsBodyExecution.java

```diff
@@ -194,15 +195,21 @@ private Continuable createContinuable(CpsThread currentThread, CpsCallableInvoca
         t.getExecution().runInCpsVmThread(new FutureCallback<CpsThreadGroup>() {
             @Override public void onSuccess(CpsThreadGroup g) {
                 try {
+                    if (g.threads.size() == 1) {
```

Comment:
I can't optimize this with the block lookup cache since the body isn't necessarily enclosing (boo), but at least can make it faster in the common case.